contrib/platform/test/com/sun/jna/platform/win32/WevtapiTest.java

```diff
@@ -85,16 +85,22 @@ public void testReadEvents() throws Exception {
                 IntByReference propertyCount = new IntByReference();
                 Winevt.EVT_VARIANT evtVariant = new Winevt.EVT_VARIANT();
                 for (int i = 0; i < returned.getValue(); i++) {
-                    evtHandle.setPointer(eventArray.share(i * POINTER_SIZE));
-                    buff = evtRender(buff, contextHandle, evtHandle.getValue(),
-                            Winevt.EVT_RENDER_FLAGS.EvtRenderEventValues, propertyCount);
-                    useMemory(evtVariant, buff, 0);
-                    assertThat("Provider Name", evtVariant.field1.StringVal, is("testSource"));
-                    sb.append(evtVariant.field1.StringVal);
-                    useMemory(evtVariant, buff, 1);
-                    assertThat("EventRecordID", evtVariant.field1.UInt64Val, is((long) arrayIndex * eventArraySize + i + 1));
-                    useMemory(evtVariant, buff, 2);
-                    assertThat("EventID", evtVariant.field1.UInt64Val, is((long) arrayIndex * eventArraySize + i + 1));
+                    try {
+                        evtHandle.setPointer(eventArray.share(i * POINTER_SIZE));
+                        buff = evtRender(buff, contextHandle, evtHandle.getValue(),
+                                Winevt.EVT_RENDER_FLAGS.EvtRenderEventValues, propertyCount);
+                        useMemory(evtVariant, buff, 0);
+                        assertThat("Provider Name", evtVariant.field1.StringVal, is("testSource"));
+                        sb.append(evtVariant.field1.StringVal);
+                        useMemory(evtVariant, buff, 1);
+                        assertThat("EventRecordID", evtVariant.field1.UInt64Val, is((long) arrayIndex * eventArraySize + i + 1));
+                        useMemory(evtVariant, buff, 2);
+                        assertThat("EventID", evtVariant.field1.UInt64Val, is((long) arrayIndex * eventArraySize + i + 1));

```

Comment:
Why does this even work? EventId is in the range > 5000 in your test events. If I'm not mistaken, your reading the wrong field here and just because you read the field directly before (with the correct value).